cpu/fe310/start.S

```diff
@@ -43,8 +43,6 @@ _start:
 2:
 
     /* Call global constructors */
-    la a0, __libc_fini_array
```

Comment:
I'm wondering if this shouldn't be in its own PR. Using just this change, it saves some bytes on ROM compared to master. And I didn't see any functional regression but I haven't tested in depth (no compile_and_test, just hello-world and default apps).

Reply:
It saves memory on all systems, but it's required for picolibc as picolibc has per-thread destructors and this is executed out of thread context (hence there isn't any TLS available here). I guess I'd like to know what global destructors might be ever be registered and if they are expected to be called when exit() is called?